core/DataTable.php

```diff
@@ -204,6 +204,14 @@ class DataTable implements DataTableInterface, \IteratorAggregate, \ArrayAccess
     const LABEL_SUMMARY_ROW = -1;
     const LABEL_TOTALS_ROW = -2;
 
+    /**
+     * The ID of the special metadata row. This row only exists in the serialized row data and stores the datatable metadata.
+     *
+     * This allows us to save datatable metadata in archive data.
+     */
+    const ID_ARCHIVED_METADATA_ROW = -3;
+    const LABEL_ARCHIVED_METADATA_ROW = -3;
```

Comment:
could it be an issue when -3 is actually a value? Like usually so far we don't archive the totals row for example so there it was maybe not an issue? Similar to ranking query wondering should that maybe be like "___metadatarow__archive__"?

Reply:
👍 makes sense.

This would also be a problem w/ the summary row as well... actually, rows are associated by ID, not label, so there might be some issues in the future, but perhaps not now. I'll use the better label regardless.

Reply:
@diosmosis be good to make the change 👍  BTW: You maybe also want to move it up next to the other `LABEL*` constants

Reply:
@tsteur Updated.